packages/web-worker/src/test/e2e.test.ts

```diff
@@ -520,6 +520,231 @@ describe('web-worker', () => {
+      const {workspace, browser} = context;
+
+      await workspace.write(
+        mainFile,
+        `
+        import {createWorker, terminate} from '@shopify/web-worker';
+        self.worker = createWorker(() => import('./worker'))();
+
+        (async () => {
+          await terminate(self.worker);
+        })();
+      `,
+      );
+
+      await workspace.write(
+        workerFile,
+        `
+        export function greet() {
+          return 'Hi, friend!';
+        }
+      `,
+      );
+
+      await runWebpack(context);
+
+      const page = await browser.go();
+      expect(page.workers()).toHaveLength(0);
+    });
+  });
+
+  it('throws an error when calling a function on a terminated worker from the main thread', async () => {
+    const greetingPrefix = 'Hello ';
+    const greetingTarget = 'world';
+    const testId = 'WorkerResult';
+
+    await withContext('error-on-terminated-worker-calls', async context => {
+      const {workspace, browser} = context;
+
+      await workspace.write(
+        mainFile,
+        `
+          import {createWorker, terminate} from '@shopify/web-worker';
+          self.worker = createWorker(() => import('./worker'))();
+          (async () => {
+            await terminate(self.worker);
+            let result;
+            try {
+              result = await self.worker.greet(${JSON.stringify(
+                greetingTarget,
+              )});
+            } catch (error){
+              result = error.toString();
+            }
+            const element = document.createElement('div');
+            element.setAttribute('id', ${JSON.stringify(testId)});
+            element.textContent = result;
+            document.body.appendChild(element);
+          })();
+        `,
+      );
+
+      await workspace.write(
+        workerFile,
+        `
+          export function greet(name) {
+            return \`${greetingPrefix}\${name}\`;
+          }
+        `,
+      );
+
+      await runWebpack(context);
+
+      const page = await browser.go();
+      const workerElement = await page.waitForSelector(`#${testId}`);
+      const textContent = await workerElement.evaluate(
+        element => element.innerHTML,
+      );
+      expect(textContent).toBe(
+        'Error: You attempted to call a function on a terminated web worker.',
+      );
+    });
+  });
+
+  it('releases memory when the worker is terminated', async () => {
+    const testId = 'WorkerResult';
+
+    await withContext('terminated-worker-releases-memory', async context => {
+      const {workspace, browser} = context;
+
+      await workspace.write(
+        mainFile,
+        `
+          import {createWorker, terminate} from '@shopify/web-worker';
+
+          self.prepare = () => {
+            start();
+
+            self.func = () => {};
+            self.worker = createWorker(() => import('./worker'))();
+
+            self.WorkerTestClass = class WorkerTestClass {constructor(id) {this.id = id;}}
```

Comment:
Follow the more modern examples for setting up this style of test, this is using a slightly sloppier setup I had earlier I think?

Reply:
Sorry, what _more modern examples_ are you referring to?

Reply:
See the other memory management tests, they do their setup more succinctly

Reply:
Updated